Add frontmatter to all doc pages

beepsboops opened the pull request:
As per Adi's request, I added frontmatter to each doc page for the Global Sync Guide i.e. I added the following to the top of each page.

---
title: RadiantOne Global Synchronization Guide
description: RadiantOne Global Synchronization Guide
---

pnolan89 commented on documentation/global-sync-guide/TOC.md:
```diff
@@ -1,3 +1,8 @@
+---
+title: RadiantOne Global Synchronization Guide
+description: RadiantOne Global Synchronization Guide
```
Shouldn't the description be different than the title? I'm assuming these are going to be used for SEO, so we need at least _something_ in the description that gives a bit more context to what's contained in the doc. Doesn't have to be more than 1-2 sentences.

pnolan89 reviewed (approved):
Can we please just update the `title` and `description` to match the H1 of each doc? We will revisit the `description` metadata to make it more functional at a later date if Radiant wants to prioritize it for 7.4.

beepsboops commented:
I updated all headings to sentence case and updated all links to match this new case. I also updated the `title` and `description` to match the H1 of each doc.
